rag/app/naive.py

```
@@ -127,6 +127,72 @@ def by_deepdoc(filename, binary=None, from_page=0, to_page=MAXIMUM_PAGE_NUMBER,
+
+    Thin wrapper around :func:`common.parser_config_utils.has_mineru_options`
+    kept here so callers in this module don't need to import the helper
+    separately. Used by :func:`_dispatch_pdf_parser` to recover from a
+    misconfigured ``layout_recognize`` value (a stale TenantModel id
+    rather than the ``"MinerU"`` keyword) — see issue #17114.
+    """
+    from common.parser_config_utils import has_mineru_options
+    return has_mineru_options(parser_config)
+
+
+def _dispatch_pdf_parser(parser_config: dict, opendataloader_llm_name=None) -> tuple[Callable, str, Any, str, Any]:
```

Ruff (0.16.2)

[error] Undefined name `Callable`

(F821)